### PR TITLE
PMC and Whiteout stuff

### PR DESCRIPTION
**Pull request by CapCamIII**
# About the pull request

As a preword, I came up with every change in this PR and coded them in the span of 10 hours, so some things may be iffy.

PMCs and Whiteout may be good, but they're a bit outdated, this modernizes multiple loadout aspects, enables antag vendors for admin-spawn, and does some balance changes to certain portions(specifically chem ERT).

Individual changes, numbered, will go over each in why its good, may have forgotten one or two things.

1. Buffs whiteout flamer with blue flame damage, belt-linked magharn, and pyro underbarrel extinguisher
2. Adds a synthetic repair kit(medkit), with synth repair tools, gives it to whiteout.
3. Adds breaching charges and swaps crowbars to tactical in whiteout loadouts
4. Gives whiteout PMC sniper goggles(thermals)
5. Gives whiteout medic the required gear to actually repair a downed member of the team, and just a lot of synth heals.
6. Gives whiteout leader a pyro extinguisher.
7. Gives whiteout weapons default attachments.
8. Adds an 'advanced' mini-flamer with the same stats as UPP integrated UBF, gives it to NSG23 and random m41/2 attachie.
9. Gives detainer PMCs a version of the corporate m41A MK2(goon gun), with attachies and adv flamer to replace flamethrower.
10. Swaps chem PMC TL P90(name is too long) with an M41/2
11. Gives normal PMC ERT roles a webbing vest with meds and miniextinguisher
12. Gives PMC Surgeon the essentials required to act as a medic, swaps NSG with M39/2, gives them a normal but unique look.
13. Whiteout and PMC SG powerpacks have 30k power by default, instead of 10k.
14. Gives PMC guns more options for random attachments, and gives m41/2 its intended collapsible stock
15. Gives PMC engi m39/2 instead of P90
16. Removes flamer from potential PMC loadouts.
17. Gives PMCs better CQC skill.
18. Gives PMC TLs autoinjector pouches(which gives chem TL a second pouch in the first place)
19. Detainer PMCs now have tac-sechuds, PMC TLs have sensorhuds.
20. **Enables antag vendor for PMC roles.** (Look at file changes for the specific things, too long for pr desc)

# Explain why it's good for the game

Modernization and some needed loadout/balance changes(IMO).
Per number:

1. Whiteout flamer did worse damage than napalm, and was incredibly easy to lose.
2. More heals for each member of the whiteout team, allows more self-sufficiency.
3. Breach charges for tacticool breaches, crowbar doubles as a melee weapon in case the player doesn't know about synth punch
4. Whiteout didn't have proper NVG, thermals allows them to do tacticooler breaches by lining up people wth breach charge.
5. Whiteout medic didn't have a lot of heals at all, and didn't have a defib, so they weren't much of a medic.
6. To extinguish the flames and lead charges, works like pyro underbarrel extinguisher, but handheld.
7. Default tacticool attachment, already made whiteout subtypes for HEAP, why not give them good attachies with them too.
8. Mini-flamer sucks, gives a better version for NSG and m41/2, same stats as the already existing UPP integrated UBF.
9. Detainers had flamers which sucked, gives them corpo m41s which aren't as good as /2s, but have adv UBF for flames.
10. P90 sucks, having default m41/2 fits with other leader type, also gives them a better gun than their underlings.
11. Medkit but in a webbing, its my personal combat webbing load when I play so its good.
12. PMC Surgeon was horrifically undergeared, they didn't even have a medhud, gives them basic gear similar to PMC med but with a beret to tell them apart.
13. ERTs don't have spare batteries to get usually, more staying power in fights.
14. More options for attachies to further make PMC weapons better, also gives m41/2 the m41 collapsible stock because it needed it.
15. P90 sucks, support roles getting the m39/2 is cool.
16. Flamer sucks to get as PMC, and adv. UBF as a potential m41/2 attachie makes a full-sized flamer useless too.
17. PMCs could fireman carry, and multi-strip, but did it horrifically slow, gives them MP level CQC(master for spec, expert for TL)
18. TL getting extra meds is neat, also chem TL had an empty pocket slot and no meds, thats bad.
19. Sechud-thing for PMC detainers is useful(stops flashbang trolling for one), giving PMC TL a sensorhud to watch their team's status is also good.
20. Makes doing event bases for PMCs much easier, too long to post the specific changes here but look at files for them.

# Testing Photographs and Procedure

I forgot to take screenshots but it all works. 👍 
I spawned myself as every changed role and tested every individual change extensively.

# Changelog

:cl:
add: PMCs are now able to use antag vendors.
balance: Multiple loadout and skill changes to PMCs and Whiteout
balance: Buffs whiteout flame damage to blue flame damage.
fix: PMC Investigator Lead now appropriately spawns with a medical pouch in their pocket, instead of nothing.
fix: Maximum skill preset now appropriately also has BE and intel skill, at maximum of course.
fix: PMC Smartgunner now appropriately a VP78 to go along with their VP78 magazines
fix: Whiteout now appropriately have night vision.
fix: M41A/2 now appropriately comes equipped with a collapsible stock.
spellcheck: PMC smartgun drum now has a seperate description and name from base SG to match its different appearance.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun_attachables.dm [hunk 1]
```diff
@@ -2073,6 +2073,14 @@ Defined in conflicts.dm of the #defines folder.
 	burn_level = BURN_LEVEL_TIER_5
 	burn_duration = BURN_TIME_TIER_2
 
+/obj/item/attachable/attached_gun/flamer/advanced//essentially a better version of mini-flamer for PMCs, copy of integrated but is removable
```
Comment (carlarctg) [inline code/modules/projectiles/gun_attachables.dm:2076]: if it's the same as the integrated flamethrower it should be a subtype. also it should only be added for pmcs, survivors should not be able to find these

File: code/modules/projectiles/guns/flamer/flamer.dm [hunk 2]
```diff
@@ -233,10 +233,20 @@
...
+		/obj/item/attachable/magnetic_harness,
+		/obj/item/attachable/attached_gun/extinguisher
+	)
+	starting_attachment_types = list(/obj/item/attachable/attached_gun/extinguisher/pyro, /obj/item/attachable/magnetic_harness)
 	flags_equip_slot = SLOT_BACK | SLOT_WAIST
+	auto_retrieval_slot = WEAR_WAIST
 	current_mag = /obj/item/ammo_magazine/flamer_tank/EX
+	flags_gun_features = GUN_WY_RESTRICTED|GUN_WIELDED_FIRING_ONLY
+
+/obj/item/weapon/gun/flamer/deathsquad/nolock
```
Comment (carlarctg) [inline code/modules/projectiles/guns/flamer/flamer.dm:248]: really man.

File: code/modules/reagents/chemistry_reagents/other.dm [hunk 3]
```diff
@@ -763,6 +763,20 @@
 		PROPERTY_FIRE_PENETRATING	= 1
 	)
 
+/datum/reagent/napalm/deathsquad //version of fuel for dsquad flamers.
```
Comment (Unknownity) [inline code/modules/reagents/chemistry_reagents/other.dm:766]: What is precisely the point of adding an entirely new chemical if the previous one is only used by whiteout flamers?
This is simply adding unnecessary code bloat, just change the values of the original.

Comment (CapCamIII) [reply code/modules/reagents/chemistry_reagents/other.dm:766]: To allow the old version to be separately used and balanced as a fuel-type.  If a dev tells me I'll remove it though 😄

File: code/modules/projectiles/gun_attachables.dm (hunk 1 above)
Comment (CapCamIII) [reply code/modules/projectiles/gun_attachables.dm:2076]: Oh yeah I forgot to do that, will do it now.
👍

File: code/modules/projectiles/guns/flamer/flamer.dm (hunk 2 above)
Comment (CapCamIII) [reply code/modules/projectiles/guns/flamer/flamer.dm:248]: Do you want the admin minor event spawning a baldie as dsquad who dies and gives marines the omega-giga-flamer from hell.

File: code/modules/projectiles/gun_attachables.dm [hunk 4]
```diff
@@ -2073,6 +2073,9 @@ Defined in conflicts.dm of the #defines folder.
 	burn_level = BURN_LEVEL_TIER_5
 	burn_duration = BURN_TIME_TIER_2
 
+/obj/item/attachable/attached_gun/flamer/integrated/advanced//different name, same stats, intended to be used as a removable version
```
Comment (stanalbatross) [inline code/modules/projectiles/gun_attachables.dm:2076]: this shouldn't be subtyped off "integrated", pathing should be like advanced and advanced/integrated

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by stanalbatross** [comment]
`Removes flamer from potential PMC loadouts.`
highly non movielike...

**Comment by Nanu308** [comment]
> PMCs could fireman carry, and multi-strip, but did it horrifically slow, gives them MP level CQC(master for spec, expert for TL)

Drop this, rest is fine.

File: code/modules/gear_presets/pmc.dm [hunk 5]
```diff
@@ -373,13 +1010,91 @@
...
+	H.equip_to_slot_or_del(new /obj/item/clothing/accessory/storage/black_vest, WEAR_ACCESSORY)
+	H.equip_to_slot_or_del(new /obj/item/stack/medical/bruise_pack, WEAR_IN_ACCESSORY)
+	H.equip_to_slot_or_del(new /obj/item/stack/medical/bruise_pack, WEAR_IN_ACCESSORY)
+	H.equip_to_slot_or_del(new /obj/item/stack/medical/splint, WEAR_IN_ACCESSORY)
+	H.equip_to_slot_or_del(new /obj/item/tool/extinguisher/mini, WEAR_IN_ACCESSORY)
+	H.equip_to_slot_or_del(new /obj/item/cell/hyper, WEAR_IN_ACCESSORY)
+	H.equip_to_slot_or_del(new /obj/item/smartgun_powerpack/pmc, WEAR_BACK)
 	H.equip_to_slot_or_del(new /obj/item/weapon/gun/smartgun/dirty, WEAR_J_STORE)
-	H.equip_to_slot_or_del(new /obj/item/storage/pouch/magazine/pistol/pmc_vp78, WEAR_R_STORE)
+	H.equip_to_slot_or_del(new /obj/item/storage/pouch/magazine/pistol/pmc_mateba, WEAR_R_STORE)
```
Comment (Nanu308) [inline code/modules/gear_presets/pmc.dm:1021]: Drop this as well, keep the VP78

**Review by Nanu308 — CHANGES_REQUESTED** [review]

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Two things:
Make a new zip tie box or revert the medium size changes.
Mateba/deagle high impact ammo should really be reserved for those that can BE (CO/staff).

**Comment by CapCamIII** [comment]
PC is dead, has been since the ninth, gotta replace it as it can’t be fixed. Don’t know when that’ll be so closing as unable to work on this.

**Comment by CapCamIII** [comment]
We are so back.

**Comment by Nanu308** [comment]
If need be balance can be tweaked further down the line, but check out where hvh variants of stuff is used if any. From memory it's old and unmaintained andshould probably be deleted. tldr. look into /hvh stuff and remove it unless good reason to stay

**Review by harryob — APPROVED** [review]

**Pull request merged**